### PR TITLE
bpo-19072: Make @classmethod support chained decorators

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://bugs.python.org/issue19072

### CODE REVIEW

File: Objects/funcobject.c [hunk 1]
```diff
@@ -703,6 +703,10 @@ cm_descr_get(PyObject *self, PyObject *obj, PyObject *type)
     }
     if (type == NULL)
         type = (PyObject *)(Py_TYPE(obj));
+    if (Py_TYPE(cm->cm_callable)->tp_descr_get != NULL) {
```
Comment (berkerpeksag) [inline Objects/funcobject.c:744]: We probably need to a comment to explain why we added this.

File: Misc/NEWS.d/next/Core and Builtins/2018-07-23-13-09-54.bpo-19072.Gc59GS.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The :class:`classmethod` decorator now works correctly when it was wrapped by
```
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Core and Builtins/2018-07-23-13-09-54.bpo-19072.Gc59GS.rst]: I'm not sure this NEWS entry is descriptive, so I'm open to suggestions.

File: Lib/test/test_property.py [hunk 3]
```diff
@@ -183,6 +183,27 @@ def test_refleaks_in___init__(self):
             fake_prop.__init__('fget', 'fset', 'fdel', 'doc')
         self.assertAlmostEqual(gettotalrefcount() - refs_before, 0, delta=10)
 
+    @unittest.skipIf(sys.flags.optimize >= 2,
+                     "Docstrings are omitted with -O2 and above")
+    def test_class_property(self):
+        class A:
+            @classmethod
+            @property
+            def __doc__(cls):
```
Comment (berkerpeksag) [inline Lib/test/test_property.py:192]: I'm not sure where is the correct place to document this feature.

File: Lib/test/test_property.py [hunk 4]
```diff
@@ -183,6 +183,27 @@ def test_refleaks_in___init__(self):
             fake_prop.__init__('fget', 'fset', 'fdel', 'doc')
         self.assertAlmostEqual(gettotalrefcount() - refs_before, 0, delta=10)
 
+    @unittest.skipIf(sys.flags.optimize >= 2,
+                     "Docstrings are omitted with -O2 and above")
```
Comment (serhiy-storchaka) [inline Lib/test/test_property.py:187]: Does sys.flags.optimize really affects this test?

File: Misc/NEWS.d/next/Core and Builtins/2018-07-23-13-09-54.bpo-19072.Gc59GS.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Misc/NEWS.d/next/Core and Builtins/2018-07-23-13-09-54.bpo-19072.Gc59GS.rst]: Isn't "when it **wraps** a descriptor" more correct?

**Review by serhiy-storchaka — COMMENTED** [review]
Please provide microbenchmark results.

**Pull request merged**